src/software/ai/hl/stp/play/defense_play.cpp

```diff
@@ -97,23 +116,71 @@ void DefensePlay::getNextTactics(TacticCoroutine::push_type &yield)
 
         // Assign ShadowEnemy tactics until we have every enemy covered. If there any
         // extra friendly robots, have them perform a reasonable default defensive tactic
-        for (unsigned i = 0;
-             i < std::min(stop_tactics.size(), shadow_enemy_tactics.size()); i++)
+        if (enemy_threats.size() > 0)
         {
-            if (i < enemy_threats.size())
+            defense_shadow_enemy_tactic->updateWorldParams(
+                world.field(), world.friendlyTeam(), world.enemyTeam(), world.ball());
+            defense_shadow_enemy_tactic->updateControlParams(enemy_threats.at(1));
+            result.emplace_back(defense_shadow_enemy_tactic);
+        }
+        else
+        {
+            // swarm the enemy nearest the ball
+            if (world.enemyTeam().numRobots() > 0)
             {
-                shadow_enemy_tactics.at(i)->updateWorldParams(
-                    world.field(), world.friendlyTeam(), world.enemyTeam(), world.ball());
-                shadow_enemy_tactics.at(i)->updateControlParams(
-                    enemy_threats.at(i), ROBOT_MAX_RADIUS_METERS * 3);
-                result.emplace_back(shadow_enemy_tactics.at(i));
+                Robot nearest_enemy_robot = *std::min_element(
+                    world.enemyTeam().getAllRobots().begin(),
+                    world.enemyTeam().getAllRobots().end(),
+                    [this](const Robot& r1, const Robot& r2) {
+                        return dist(r1.position(), world.ball().position()) <
+                               dist(r2.position(), world.ball().position());
+                    });
+                Point block_point =
+                    nearest_enemy_robot.position() +
+                    Point::createFromAngle(nearest_enemy_robot.orientation()) *
+                        ROBOT_MAX_RADIUS_METERS * 3;
+                move_tactics[0]->updateControlParams(
+                    block_point, nearest_enemy_robot.orientation() + Angle::half(), 0.0);
+                move_tactics[1]->updateControlParams(
+                    block_point, nearest_enemy_robot.orientation() + Angle::half(), 0.0);
+                result.emplace_back(move_tactics[0]);
+                result.emplace_back(move_tactics[1]);
             }
             else
             {
-                result.emplace_back(stop_tactics.at(i));
+                // somehow they have 0 robots
+                LOG(WARNING) << "0 enemy robots so we hit a very stupid fallback case";
```

Comment:
I get that this shouldn't happen, but maybe we should have 2 ShootGoal tactics if there are no enemy robots?

Reply:
I think that's worse because they could interfere with each other. I'd like to leave it as is for now.